src/Common/ThreadPool.cpp

```diff
@@ -205,6 +205,13 @@ size_t ThreadPoolImpl<Thread>::active() const
     return scheduled_jobs;
 }
 
+template <typename Thread>
+bool ThreadPoolImpl<Thread>::finished() const
```

Comment:
Consider to rename bool shutdownBecauseOfJobException() { return shutdown && shutdown_on_exception; }
Currently shutdown can be set to true in finizalize.

Reply:
>Consider to rename bool shutdownBecauseOfJobException() { return shutdown && shutdown_on_exception; }

`shutdownBecauseOfJobException` looks too long to me, but OK

>Currently shutdown can be set to true in finizalize.

And is this a problem to return true even if it was shutted down due to finalize() ?
The point of this method is to say that pool will not accept any new jobs anymore.

Reply:
And even with proposed change it will still be not that it means, since it shutdown_on_exception is simply a bool flag, but shutdown can be set to true anyway in finalize(), but finalize() is private and it is called only from dtor, so this proposal looks superfluous to me

Reply:
@azat sorry I misread your comment to code, maybe better to remove part of comment
```
/// (i.e. exception had already been thrown from the job callback).
```
for client it does not matter.

Reply:
Ok, I've changed the comment.